test/jdk/java/lang/annotation/typeAnnotations/GetAnnotatedReceiverType.java

```diff
@@ -132,9 +150,35 @@ public static void main(String[] args) throws NoSuchMethodException {
         checkNull(instance3.getAnonymousClass().getDeclaredConstructors()[0],
                 "getAnnotatedReceiverType() on a constructor for an anonymous class should return null");
 
+        Inner4<?> instance4 = outer.new Inner4<String>();
+        Inner4<?>.Inner5 instance5 = instance4.new Inner5();
+        Inner4<?>.Inner5.Inner6 instance6 = instance5.new Inner6();
+
+        checkAnnotatedReceiverType(instance4.getClass().getConstructors()[0], false,
+                "The type of .getAnnotatedReceiverType().getType() for this constructor should be");
+        checkAnnotatedReceiverType(instance5.getClass().getConstructors()[0], true,
+                "The type of .getAnnotatedReceiverType().getType() for this constructor should be");
+        checkAnnotatedReceiverType(instance6.getClass().getConstructors()[0], true,
+                "The type of .getAnnotatedReceiverType().getType() for this constructor should be");
+        checkAnnotatedReceiverType(outer.getClass().getMethod("method0"), false,
+                "The type of .getAnnotatedReceiverType().getType() for this method should be");
+        checkAnnotatedReceiverType(instance4.getClass().getMethod("innerMethod4"), true,
+                "The type of .getAnnotatedReceiverType().getType() for this method should be");
+        checkAnnotatedReceiverType(instance5.getClass().getMethod("innerMethod5"), true,
+                "The type of .getAnnotatedReceiverType().getType() for this method should be");
+        checkAnnotatedReceiverType(instance2.getClass().getMethod("innerMethod2"), false,
+                "The type of .getAnnotatedReceiverType().getType() for this method should be");
+        checkAnnotatedReceiverType(instance3.getClass().getMethod("innerMethod3"), false,
+                "The type of .getAnnotatedReceiverType().getType() for this method should be");
+
+        Inner2.Inner3.Inner7<?> instance7 = instance3.new Inner7<String>();
+        checkAnnotatedReceiverType(instance7.getClass().getMethod("innerMethod7"), true,
+                "The type of .getAnnotatedReceiverType().getType() for this method should be");
+        recursiveCheckAnnotatedOwnerTypes(instance7.getClass().getMethod("innerMethod7").getAnnotatedReceiverType());
+
         if (failures != 0)
             throw new RuntimeException("Test failed, see log for details");
-        else if (tests != 15)
+        else if (tests != 25)
```

Comment:
this looks a bit brittle, isn't it better to count the number of tests failing and issue an error if that number is > 0?

Reply:
Yes, not great, but at least it isn't brittle when running the test, only when changing it. I'd like to take a separate pass over the tests for 17 if possible.

Reply:
> Yes, not great, but at least it isn't brittle when running the test, only when changing it. I'd like to take a separate pass over the tests for 17 if possible.

what about declaring a static final field for that value instead of using a literal?

Reply:
Done